string path_name
string dir
---

string message
